test/handshake_test.cpp

```diff
@@ -129,6 +133,194 @@ TEST_CASE("certificates") {
   }
 }
 
+TEST_CASE("client certificates") {
+  using namespace std::string_literals;
+
+  SECTION("wintls client certificate missing with openssl server") {
+    using namespace boost::system;
+    wintls_client_context client_ctx;
+    asio_ssl_server_context server_ctx;
+    server_ctx.enable_client_verify();
+
+    net::io_context io_context;
+    boost::wintls::stream<test_stream> client_stream(io_context, client_ctx);
+    boost::asio::ssl::stream<test_stream> server_stream(io_context, server_ctx);
+
+    client_stream.next_layer().connect(server_stream.next_layer());
+
+    auto client_error = errc::make_error_code(errc::not_supported);
+    client_stream.async_handshake(boost::wintls::handshake_type::client,
+                                  [&client_error](const boost::system::error_code& ec) {
+                                    client_error = ec;
+                                  });
+
+    auto server_error = errc::make_error_code(errc::not_supported);
+    server_stream.async_handshake(asio_ssl::stream_base::server,
+                                  [&server_error](const boost::system::error_code& ec) {
+                                    server_error = ec;
+                                  });
+    io_context.run();
+    // client handshake is failed by server
+    CHECK(client_error);
+    // Note: error code mapping is not working, so compare string.
```

Comment:
What does it mean that "error code mapping is not working"? Could it be made to work?

Reply:
I added more info in the new version. It seems like the top bits of the error code is used for something else. I will need to debug into the openssl code further to find out why, but it is out of scope of this PR.